synapse/config/server.py

```diff
@@ -621,6 +627,13 @@ def generate_config_section(
         #
         #require_auth_for_profile_requests: true
 
+        # Whether to require a user to share a room with another user in order
+        # to retrieve their profile information. Only checked on Client-Server
+        # requests. Profile requests from other servers should be checked by the
+        # requesting server. Defaults to 'false'.
+        #
+        #limit_profile_requests_to_users_who_share_rooms: false
```

Comment:
These days we're trying to do "uncomment to switch the functionality".

```suggestion
        #limit_profile_requests_to_users_who_share_rooms: true
```